performance.lic

```diff
@@ -52,37 +63,64 @@ class Performance
       when /muster the strength to perform/
         pause 30
       when /You begin (.*)/, /You continue (.*)/, /You find yourself relaxing (.*)/, /You concentrate on your sense of rhythm (.*)/, /You cannot use the/
-        exit if done_training?
+        if done_training?
+          wear_hand_armor_and_put_away_instrument
+          exit
+        end
       when /You finish (.*)/
         if done_training?
+          wear_hand_armor_and_put_away_instrument
           exit
         else
-          play_song?(@settings, @song_list, is_instrument_worn, skip_clean)
+          DRC.play_song?(@settings, song_list, is_instrument_worn, skip_clean)
         end
       when /Perhaps you should find somewhere drier/
-        exit
+        if done_training?
+          wear_hand_armor_and_put_away_instrument
+          exit
+        end
       end
     end
   end
 
+  def remove_hand_armor_and_get_instrument
+    if DRCI.remove_item?(@hand_armor)
+      @hand_armor_removed = true
+      DRCI.stow_item?(@hand_armor)
+    end
+    if DRCI.get_item?(@instrument)
```

Comment:
Won't this try and get an instrument even if it shouldn't be removed (e.g. zills)? I don't see any guards in here